### ISSUE TITLE
Regression in version 1.35.0: E3002 Additional properties are not allowed ('Fn::Transform' was unexpected)

### ISSUE THREAD

**Issue by rhopman** [aws-cloudformation/cfn-lint#4136]
### CloudFormation Lint Version

1.35.0

### What operating system are you using?

Debian (wsl2)

### Describe the bug

There appears to be some regression in version 1.35.0. A template that is using `Fn::Transform` is giving an error `E3002 Additional properties are not allowed ('Fn::Transform' was unexpected)` since version 1.35.0. In version 1.34.2 it was parsed correctly, and CloudFormation deploys it without complaining.

Note that an error is only shown for the second `Fn::Transform` in the reproduction template. Maybe this is because it is in a list? Just a guess.

### Expected behavior

Template should be parsed without error.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09

Resources:
  ManagedLoginBranding:
    Type: AWS::Cognito::ManagedLoginBranding
    Properties:
      UserPoolId: XXXXXX
      ClientId: YYYYYY
      UseCognitoProvidedValues: false
      Settings:
        Fn::Transform:
          Name: AWS::Include
          Parameters:
            Location: settings.json
      Assets:
        - Category: FAVICON_SVG
          ColorMode: LIGHT
          Extension: SVG
          Fn::Transform:
            Name: AWS::Include
            Parameters:
              Location: favicon.json
```
